File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_129/javascript/solution.js
function minPath(grid, k) {
  const n = grid.length;
  const DIRECTIONS = [[0, 1], [0, -1], [1, 0], [-1, 0]];
  
  // Find the position of value 1
  let row = 0, col = 0;
  for (let i = 0; i < n; i++) {
    const j = grid[i].indexOf(1);
    if (j !== -1) {
      row = i;
      col = j;
      break;
    }
  }
  
  const path = [1];
  
  // Build path of length k
  for (let step = 1; step < k; step++) {
    let minVal = Infinity;
    let nextRow = row, nextCol = col;
    
    for (const [dr, dc] of DIRECTIONS) {
      const newRow = row + dr;
      const newCol = col + dc;
      
      if (newRow >= 0 && newRow < n && newCol >= 0 && newCol < n) {
        const val = grid[newRow][newCol];
        if (val < minVal) {
          minVal = val;
          nextRow = newRow;
          nextCol = newCol;
        }
      }
    }
    
    row = nextRow;
    col = nextCol;
    path.push(grid[row][col]);
  }
  
  return path;
}

File: comparison/results/haiku/code-simplifier/humaneval/HumanEval_129/go/solution.go
func minPath(grid [][]int, k int) []int {
	n := len(grid)

	// Find position of 1
	var startR, startC int
	for i := 0; i < n; i++ {
		for j := 0; j < n; j++ {
			if grid[i][j] == 1 {
				startR, startC = i, j
				break
			}
		}
	}

	result := []int{1}
	r, c := startR, startC

	directions := [][2]int{{-1, 0}, {1, 0}, {0, -1}, {0, 1}}

	for len(result) < k {
		minVal := n*n + 1
		minR, minC := r, c

		for _, dir := range directions {
			nr, nc := r+dir[0], c+dir[1]
			if nr >= 0 && nr < n && nc >= 0 && nc < n && grid[nr][nc] < minVal {
				minVal = grid[nr][nc]
				minR, minC = nr, nc
			}
		}

		result = append(result, minVal)
		r, c = minR, minC
	}

	return result
}
